scripts/xcode_local_build

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+set -e
+set -x
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+PROJECT_DIR="$SCRIPT_DIR/../"
+
+pushd $PROJECT_DIR
+
+rm -rf ./LibNoosphere.xcframework
+
+$SCRIPT_DIR/generate_headers
+
+TARGETS=(
+    "aarch64-apple-ios"
+    "x86_64-apple-ios"
+    "aarch64-apple-ios-sim"
+    "x86_64-apple-darwin"
+    "aarch64-apple-darwin"
+)
+
+for TARGET in "${TARGETS[@]}"; do
+    rustup target install $TARGET
+    cargo build --package noosphere --release --target $TARGET --locked
+    # cargo build --package noosphert buie --target $TARGET --locked
+done
+
+mkdir -p ./target/macos
+mkdir -p ./target/simulator
+
+lipo -create \
+    ./target/x86_64-apple-darwin/release/libnoosphere.a \
+    ./target/aarch64-apple-darwin/release/libnoosphere.a \
+    -output ./target/macos/libnoosphere.a
+
+lipo -create \
+    ./target/x86_64-apple-ios/release/libnoosphere.a \
+    ./target/aarch64-apple-ios-sim/release/libnoosphere.a \
+    -output ./target/simulator/libnoosphere.a
+
+xcodebuild -create-xcframework \
+    -library ./target/macos/libnoosphere.a \
+    -headers ./target/headers/include/ \
+    -library ./target/simulator/libnoosphere.a \
+    -headers ./target/headers/include/ \
+    -library ./target/aarch64-apple-ios/release/libnoosphere.a \
+    -headers ./target/headers/include/ \
+    -output ./LibNoosphere.xcframework
+
+popd
+
+set +x
+set +e
```

Comment:
```suggestion
```
You don't need to leave the directory or unset your flags. Bash scripts run as a subprocess and don't affect the outer scope. This would only be important if you were writing some library that gets sourced (which runs in your currents scope). This shouldn't be that.